02week/pigLatin.js

```diff
@@ -7,13 +7,67 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+//check if it's a string of letters, function isString(), function hasOnlyLetters() - if so, turn string into array, .split()
+//if first letter is vowel, return word + 'yay', function findFirstVowel()
+//other wise, find the first vowel, function firstLetterVowel(), return the index of the vowel, letters before vowel become new (var) middle, vowel and everything following become new (var) beginning, slap an 'ay' after those bad boys
+//.join() to bring back to string
 
-function pigLatin(word) {
+const vowel = ['a','e','i','o','u'];
+const isString = (word) => {
+  return typeof(word)=='string';
+}
+
+const hasOnlyLetters=(word)=>{
+  const letter = ' abcdefghijklmnopqrstuvwxyz';
+  for(let n=0;n<word.length;n++){
+    let isLetter = false;
+    for(let m=0;m<letter.length;m++){
+      if(word[n]==letter[m]){
+      isLetter=1;
+      }
+    }
+      if(isLetter==false){
+        return false;
+      }
+  }
+  return true;
+}
 
-  // Your code here
+const findFirstVowel=(word)=>{
+  for(let y=1;y<word.length;y++){
+    for(let x=0;x<vowel.length;x++){
+      if(word[y] == vowel[x]){
+        return y
+      }
+    }
+  }
+}
 
+const firstLetterVowel = (word) => {
+  for(let x=0;x<vowel.length;x++){
+    if(word[0] == vowel[x]){
+      return true
+    }
+  }
 }
 
+const pigLatin = (word)=> {
+  word = word.trim().toLowerCase();
```

Comment:
we want to try and avoid mutating data, so store the formatted word into a new const and use that